cmake/Modules/DeprecationHelpers.cmake

```diff
@@ -0,0 +1,78 @@
+function(install_obs_plugin_with_data)
+  message(
+    DEPRECATION
+      "OBS: The install_obs_plugin_with_data command is deprecated and will be removed soon - use 'setup_plugin_target' instead."
+  )
+  _install_obs_plugin_with_data(${ARGV})
+endfunction()
+
+function(install_obs_plugin)
+  message(
+    DEPRECATION
+      "OBS: The install_obs_plugin command is deprecated and will be removed soon - use 'setup_plugintarget' instead."
+  )
+  _install_obs_plugin(${ARGV})
+endfunction()
+
+function(install_obs_datatarget)
+  message(
+    DEPRECATION
+      "OBS: The install_obs_datatarget function is deprecated and will be removed soon - use 'setup_target_resources' instead."
+  )
+  _install_obs_datatarget(${ARGV})
+endfunction()
+
+function(__deprecated_var var access)
+  if(access STREQUAL "READ_ACCESS")
+    message(DEPRECATION "OBS: The variable '${var}' is deprecated!")
+  endif()
+endfunction()
+
+function(__deprecated_feature var access)
+  if(access STREQUAL "UNKNOWN_READ_ACCESS")
+    message(
+      DEPRECATION
+        "OBS: The feature enabled by '${var}' is deprecated and will soon be removed from OBS."
+    )
+  endif()
+endfunction()
+
+set(_DEPRECATED_VARS
+    zlibPath
+    vulkanPath
+    SwigPath
+    PythonPath
+    mbedtlsPath
+    LuajitPath
+    x264Path
+    VlcPath
+    VLCPath
+    speexPath
+    rnnoisePath
+    LibfdkPath
+    curlPath
+    JanssonPath
+    FFmpegPath
+    DepsDir
```

Comment:
I've never seen this used in OBS.

Reply:
Look in the finders e.g. for FFmpeg - I dunno how many people actually used those, but they were implemented.

My thinking was, if they are implemented and now removed, we should at least give the deprecation warning (as configuration will not immediately fail, but could lead to unexpected consequences).

Reply:
I mean specifically "DepsDir".  I can't find it.  Could you point me to where it was used?

Reply:
Yeah in that case that could've been a Freudian slip.